synapse/handlers/message.py

```diff
@@ -41,6 +43,14 @@
 logger = logging.getLogger(__name__)
 
 
+class PurgeStatus(object):
+    def __init__(self):
+        self.status = "active"
+
+    def asdict(self):
+        return self.__dict__
```

Comment:
I'm not entirely convinced this is better than just using a regular dict. Having a dedicated `PurgeStatus` class is useful for documenting and enforcing the return values, but I don't think that this particular construct really does either of those. 

I'd prefer either a subclass of namedtuple (it does helpfully provide an _asdict already), or have `asdict` be explicit about the properties it returns, i.e. `{"status": self.status}`

Reply:
Also, a short docstring would be good.

Reply:
I tried a namedtuple to start with, but I wanted the individual fields to be mutable so that didn't work.

So yes, I can make it a more explicit dict.

Reply:
> I tried a namedtuple to start with, but I wanted the individual fields to be mutable so that didn't work.

Ah, of course.

Reply:
have made `asdict` more explicit, and turned `status` into a more enummy thing while I'm at it